### PR TITLE
Move sap related datasource into new insights.specs.datasource module

### PR DESCRIPTION
**Pull request by xiangce**
Signed-off-by: Xiangce Liu <xiangceliu@redhat.com>

### All Pull Requests:

Check all that apply:

* [x] Have you followed the guidelines in our Contributing document, including the instructions about commit messages?
* [ ] Is this PR to correct an issue?
* [x] Is this PR an enhancement?

### Complete Description of Additions/Changes:

Move  the existing sap datasources to the new `insights.specs.datasource` module

### CODE REVIEW

File: insights/specs/datasources/sap.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+          ]
+
+    .. note::
+        Currently, only Sap users are supported.
+    """
+    ctx = broker[HostContext]
+    llds = []
+    for sid in broker[sap_sid]:
+        usr = '{0}adm'.format(sid)
+        ret, vvs = ctx.shell_out("/bin/su -l {0} -c /bin/env".format(usr), keep_rc=True)
```
Comment (bfahr) [inline insights/specs/datasources/sap.py:84]: @xiangce you need to set a timeout value for this command.  See #3119 file insights/specs/datasources/__init__.py line 44.

File: docs/custom_datasources_index.rst [hunk 2]
```diff
@@ -9,4 +9,12 @@ insights.specs.datasources.cloud_init
 .. automodule:: insights.specs.datasources.cloud_init
     :members: cloud_cfg, LocalSpecs
     :show-inheritance:
-    :undoc-members:
\ No newline at end of file
+    :undoc-members:
+
+insights.specs.datasources.sap
+-------------------------------------
```
Comment (bfahr) [inline docs/custom_datasources_index.rst:15]: @xiangce please fix the length of this underline

**Review by bfahr — CHANGES_REQUESTED** [review]

File: docs/custom_datasources_index.rst (hunk 2 above)
Comment (xiangce) [reply docs/custom_datasources_index.rst]: Fixed in the last commit.

File: insights/specs/datasources/sap.py (hunk 1 above)
Comment (xiangce) [reply insights/specs/datasources/sap.py]: Done. @bfahr Thanks.

File: docs/custom_datasources_index.rst (hunk 2 above)
Comment (xiangce) [reply docs/custom_datasources_index.rst]: @bfahr - Would you please review this PR again, I updated it as per #3119. Thanks.

**Review by bfahr — APPROVED** [review]
Approved pending rebase of PR with latest master.

**Comment by xiangce** [comment]
> Approved pending rebase of PR with latest master.

Rebased to the latest master and tests passed. Thanks

**Comment by xiangce** [comment]
related to #3087

**Pull request merged**